contents/data-display/charts/weekly-temperature-chart/index.tsx

```diff
@@ -0,0 +1,148 @@
+    twoWeeksAhead.setDate(today.getDate() + 13)
+    const fourMonthsAgo = new Date()
+    fourMonthsAgo.setMonth(today.getMonth() - 3)
+
+    setIsNextDisabled(startDate > twoWeeksAhead)
+    setIsPreviousDisabled(startDate < fourMonthsAgo)
+
+    const endDate = new Date(startDate)
+    endDate.setDate(startDate.getDate() - 7)
+    const startDateString = formatDate(endDate)
+    const endDateString = formatDate(startDate)
+
+    const response = await fetch(
+      `https://api.open-meteo.com/v1/forecast?latitude=35.6785,34.6937,43.0642,26.2124&longitude=139.6823,135.5023,141.3468,127.6792&hourly=temperature_2m&timezone=Asia%2FTokyo&start_date=${startDateString}&end_date=${endDateString}`,
+      { cache: "force-cache" },
+    )
+
+    const data = (await response.json()) as Array<{
+      latitude: number
+      longitude: number
+      hourly: {
+        time: string[]
+        temperature_2m: number[]
+      }
+    }>
+
+    const formattedData = data.reduce(
+      (acc, region, index) => {
+        let locationKey: string
+        switch (index) {
+          case 0:
+            locationKey = "Tokyo"
+            break
+          case 1:
+            locationKey = "Osaka"
+            break
+          case 2:
+            locationKey = "Hokkaido"
+            break
+          case 3:
+            locationKey = "Okinawa"
+            break
+          default:
+            throw new Error("Unexpected location index")
+        }
+        region.hourly.time.forEach((time, i) => {
+          const formattedTime = time.replace("T", " ")
+          if (!acc[i]) acc[i] = { date: formattedTime } as Record<string, any>
+          acc[i][locationKey] = region.hourly.temperature_2m[i]
+        })
+        return acc
+      },
+      [] as Array<Record<string, any>>,
+    )
+
+    return formattedData
+  }, [startDate])
+
+  const handlePreviousWeek = () => {
+    setStartDate((prevDate) => {
+      const newDate = new Date(prevDate)
+      newDate.setDate(prevDate.getDate() - 7)
+      return newDate
+    })
+  }
+
+  const handleNextWeek = () => {
+    setStartDate((prevDate) => {
+      const newDate = new Date(prevDate)
+      newDate.setDate(prevDate.getDate() + 7)
+      return newDate
+    })
+  }
+
+  const endDate = new Date(startDate)
+  endDate.setDate(startDate.getDate() - 7)
+  const startDateString = formatDate(endDate)
+  const endDateString = formatDate(startDate)
+
+  return (
+    <Container m="auto" maxW="container.md" p={4}>
+      <Heading mb={6}>Weekly Temperature Chart</Heading>
+      <HStack mb={4} justifyContent="center">
+        <IconButton
+          onClick={handlePreviousWeek}
+          isDisabled={isPreviousDisabled}
+          icon={<ChevronLeftIcon />}
+        />
+        <Text>{`${startDateString} 〜 ${endDateString}`}</Text>
+        <IconButton
+          onClick={handleNextWeek}
+          isDisabled={isNextDisabled}
+          icon={<ChevronRightIcon />}
+        />
+      </HStack>
```

Comment:
We should try to use `md`, `lg` etc over numbers.
Also, is there a reason why you used the full-width `～` over the half-width `~`?
I also think we should add `variant` and `colorScheme` to make the button nicer.
The code will also look nicer if we use gap instead of `mb` and `<Center />` instead of `justifyContent`.

We can also remove the padding set in the options (metadata.json) so we don't have to do `p="md"` on the container element.

```tsx
 "options": {
    "container": {
      "p": "md"
    }
  }
```

```suggestion
    <Container maxW="9xl" gap="lg">
      <Heading as="h2">Weekly Temperature Chart</Heading>
      <Center gap="md">
        <IconButton
          onClick={handlePreviousWeek}
          isDisabled={isPreviousDisabled}
          icon={<ChevronLeftIcon />}
          variant="outline"
          colorScheme="primary"
        />
        <Text>{`${startDateString} ~ ${endDateString}`}</Text>
        <IconButton
          onClick={handleNextWeek}
          isDisabled={isNextDisabled}
          icon={<ChevronRightIcon />}
          variant="outline"
          colorScheme="primary"
        />
      </Center>
```

Reply:
Please remember to use semantic tokens over numbers as I have shown here.

Reply:
I fixed it.